### ISSUE TITLE
Changes between 1.6.7 and 1.7.4 cause AutoStructify to throw 'character_level_inline_markup' AttributeError

### ISSUE THREAD

**Issue by maxgrenderjones** [sphinx-doc/sphinx#5016]
I have reported this against recommonmark as https://github.com/rtfd/recommonmark/issues/104, but it looks so similar to #4493 (which was fixed on the sphinx side) that I'm not sure whose fix this is.

**Comment by tk0miya** [comment]
I'll take a look later.
Do you have any reproducible example for this? It is helpful for me.

**Comment by tk0miya** [comment]
I just posted #5028. I don't test it, but it will fix this issue. Could you try this please?

**Comment by tk0miya** [comment]
Note: This might be not raised with docutils-0.14. The initialization of `character_level_inline_markup` is required by docutils-0.13 or older.

**Comment by asmeurer** [comment]
I'm seeing a similar error with 

```
  File "/Users/aaronmeurer/anaconda3/lib/python3.5/site-packages/docutils/parsers/rst/directives/misc.py", line 70, in run
    'tab-width', self.state.document.settings.tab_width)
AttributeError: 'Values' object has no attribute 'tab_width'
```

**Comment by tk0miya** [comment]
I merged #5028, but it could not fix @asmeurer 's case. I made another PR #5052 to fix both of cases.
Could you try it please if you have time.

**Comment by tk0miya** [comment]
Fixed by #5052 
Thank you for reporting!
